cmd/ibazel/main.go

```diff
@@ -44,7 +44,7 @@ var overrideableBazelFlags []string = []string{
 	"--compile_one_dependency",
 	"--config=",
 	"--copt=",
-	"--curses=no",
+	"--curses=yes",
```

Comment:
It's been a while but I don't think this does what you suggest. This is the list of flags that you're able to specify. It's a bit weird that it says `no` here and that should be removed, but we don't pass that value down into bazel, we only use this to determine which flags can be passed down.

Does that make sense?

Reply:
Oh, I see. Yes, this makes sense.

Hmm - I wonder what the right way to trick bazel's `--curses=auto` into doing the right thing is then.

Reply:
I think you can pass `--curses=no` to ibazel and it should do the right thing ™️  (please tell me if it doesn't)

Reply:
Unfortuantely it doesn't - during the initial query step it appears to treat `--curses=yes` as a target? maybe? (not sure how best to debug it):

```
$ ibazel build --curses=yes //trunk/cli
++ 2023-03-03T16:31:39-0800 ++ ibazel build --curses=yes //trunk/cli
iBazel [4:31PM]: Querying for files to watch...
iBazel [4:31PM]: Bazel query failed: exit status 2
iBazel [4:31PM]: Error querying for source files: exit status 2
iBazel [4:31PM]: Bazel query failed: exit status 2
iBazel [4:31PM]: Error querying for source files: exit status 2
iBazel [4:31PM]: Building --curses=yes //trunk/cli
INFO: Invocation ID: ba834246-c44d-46ab-997d-4cf1eb429bce
INFO: Streaming build results to: http://buildbuddy.trunk.io/invocation/ba834246-c44d-46ab-997d-4cf1eb429bce
INFO: Analyzed target //trunk/cli:cli (286 packages loaded, 21841 targets configured).
INFO: Found 1 target...
[18 / 581] checking cached actions
```

(Also, I want `--curses=yes` behavior, not `--curses=no` behavior. It's defaulting to `--curses=no` behavior right now, presumably because since ibazel is wrapping bazel, `--curses=auto` sees that `bazel` does not have an interactive TTY and therefore resolves to `--curses=no`.)

Reply:
What happens if you change this line to be `--curses=`? I think it, with the value specified the way it is right now, will only pass `--curses=yes`, but if you remove the `yes` from this filter list then it will, I think, allow you to pass any value you want.

Also, you could set a value in your `.bazelrc` for this which might be an alternative strategy

Reply:
I'm not sure if I understand what you mean when you say

> if you remove the `yes` from this filter list then it will, I think, allow you to pass any value you want.

```
++ 2023-03-07T12:28:46-0800 ++ ibazel build --curses= //trunk/cli
iBazel [12:28PM]: Querying for files to watch...
iBazel [12:28PM]: Bazel query failed: exit status 2
iBazel [12:28PM]: Error querying for source files: exit status 2
iBazel [12:28PM]: Bazel query failed: exit status 2
iBazel [12:28PM]: Error querying for source files: exit status 2
iBazel [12:28PM]: Building --curses= //trunk/cli
ERROR: While parsing option --curses=: Not a valid --curses setting: '' (should be yes, no or auto)
INFO: Invocation ID: cf98968e-647e-4d14-b9d7-2d2c06c45ea7
INFO: Streaming build results to: http://buildbuddy.trunk.io/invocation/cf98968e-647e-4d14-b9d7-2d2c06c45ea7
iBazel [12:28PM]: Build error: exit status 2
```

Going via bazelrc does work, but I'd like to fix this so that we don't have to apply this workaround.

Reply:
Sorry to be unclear. I'm suggesting you remove `yes` from the `cmd/ibazel/main.go` file in this repo

Reply:
Ah, that makes sense!

Tested locally, works as intended.

Reply:
I wonder how we'd make curses=auto do the right thing in ibazel..

Reply:
OK, so bazel will do `--curses=yes` if both stdout/stderr are TTYs:

https://github.com/bazelbuild/bazel/blob/735539779287334ae1dcb585f5b90e902c6bf9fe/src/main/java/com/google/devtools/build/lib/runtime/UiOptions.java#L271-L277
https://github.com/bazelbuild/bazel/blob/735539779287334ae1dcb585f5b90e902c6bf9fe/src/main/cpp/option_processor.cc#L661-L662

I guess ibazel doesn't just forward stdout/stderr, but manually indirects output from those two?

Reply:
That's correct, we parse the output so that we can trigger actions in it if you configure that